### ISSUE TITLE
style binding handling css `calc` function improperly

### ISSUE THREAD

**Issue by theGivingTree** [angular/angular#21691]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

`style.paddingLeft="calc(10px + 1px * 1)` does nothing. (but `style.paddingLeft="calc(10px)` does)
`[ngStyle]="{'padding-left': 'calc(10px + 1px * 1)'}"` works as expected.

## Expected behavior

`style.paddingLeft="calc(10px + 1px * 1)` works as same as `[ngStyle]="{'padding-left': 'calc(10px + 1px * 1)'}"`

## Minimal reproduction of the problem with instructions

https://stackblitz.com/edit/angular-gde28w

## Environment

<pre><code>
Angular version: 5.x.x

Browser:
- [ ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by trotyl** [comment]
Duplicate of #15836 and #16649, function call like style value will be blocked by sanitizer.

You'll need to bypass the sanitizer and use correct binding syntax to [make it work](https://stackblitz.com/edit/angular-amkmzg?file=app/app.component.html).

**Comment by theGivingTree** [comment]
@trotyl Then, `ngStyle` was the one that doesn't work as the angular team intended.
Should I change the title or open another issue for `ngStyle`'s inconsistent behavior?

**Comment by trotyl** [comment]
Had another look of current spec:

https://github.com/angular/angular/blob/92d7060cb086632d2188c8da37ff72838d1abeef/packages/platform-browser/test/security/style_sanitizer_spec.ts#L48

This should be valid usage being supported, please continue with that.

**Comment by AHelper** [comment]
I just ran into this with a '+' in a calc string (https://plnkr.co/edit/UQsJuNRuwbpPl7cgbwlo?p=preview).  It looks like `FN_ARGS` at https://github.com/angular/angular/blob/master/packages/core/src/sanitization/style_sanitizer.ts#L32 wasn't made to handle calc strings.

Also, why aren't style bindings with ngStyle being sanitized (as styles) before getting to `Renderer2.setStyle`?  I would have expected these strings to be treated as unsafe/unsanitized style strings just as style attributes sanitize them after marking them as `SecurityContext.STYLE`...

**Comment by benlesh** [comment]
Related buganizer: https://b/115783456

**Comment by austinrmiller** [comment]
The regex were copied from closure, but those regex have since been updated to handle this: https://github.com/google/closure-library/commit/be76be0665a3e75af18403f050492d5d73e75e7c

**Comment by thePunderWoman** [comment]
This issue was closed as obsolete due to the introduction of the new `animate.enter` and `animate.leave` APIs. The `@angular/animations` package is now deprecated and will be removed in Angular v23. We hope you give the new APIs a try.
